lib/phobos.rb

```diff
@@ -31,19 +31,44 @@
 
 Thread.abort_on_exception = true
 
+# Monkey patch to fix this issue: https://github.com/zendesk/ruby-kafka/pull/732
+module Logging
+  # :nodoc:
+  class Logger
+    # :nodoc:
+    def formatter=(*args); end
+
+    # :nodoc:
+    def push_tags(*args); end
+
+    # :nodoc:
+    def pop_tags(*args); end
+  end
+end
+
 module Phobos
   class << self
     attr_reader :config, :logger
     attr_accessor :silence_log
 
     def configure(configuration)
       @config = fetch_configuration(configuration)
-      @config.class.send(:define_method, :producer_hash) { Phobos.config.producer&.to_hash }
-      @config.class.send(:define_method, :consumer_hash) { Phobos.config.consumer&.to_hash }
+      config_define_methods
       @config.listeners ||= []
       configure_logger
     end
 
+    def config_define_methods # rubocop:disable Metrics/AbcSize
+      @config.class.send(:define_method, :cache_sync_producer) do
```

Comment:
When people who read this code and our readme read the word "cache" they won't be thinking about instance variables in ruby, they will think of "caching" stuff in general which makes this name strange. It would be similar to naming `consumer_hash` to `cache_consumer_hash` or even `instance_variable_consumer_hash`.